packages/overlay/src/overlay-stack.ts

```diff
@@ -307,9 +311,15 @@ export class OverlayStack {
     }
 
     public addOverlayEventListeners(activeOverlay: ActiveOverlay): void {
-        activeOverlay.addEventListener('close', () => {
-            this.hideAndCloseOverlay(activeOverlay);
-        });
+        activeOverlay.addEventListener('close', ((
+            event: CustomEvent<OverlayCloseReasonDetail>
+        ) => {
+            this.hideAndCloseOverlay(
+                activeOverlay,
+                true,
+                !!event.detail?.reason
```

Comment:
I had to look these up - might be worth a pair of comments:

```suggestion
                true,  // animated
                !!event.detail?.reason  // clickAway
```